Samples/Widgets/MultipleWidgets/cpp-win32-packaged/SampleWidgetProviderApp/CountingWidgetImpl.cpp

```diff
@@ -0,0 +1,97 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "WidgetImpl.h"
+#include "CountingWidgetImpl.h"
+
+CountingWidget::CountingWidget(winrt::hstring const &id, winrt::hstring const& state) : m_clickCount{ 0 }, WidgetImpl(id, state)
+{
+	if (state.empty())
+	{
+		State(L"0");
+	}
+	else
+	{
+		try
+		{
+			// This particular widget stores its clickCount
+			// in the state as integer. Attempt to parse the saved state
+			// and convert it to integer.
+			m_clickCount = std::stoi(state.c_str());
+		}
+		catch (...)
+		{
+			// Parsing the state was not successfull: cached state might've been corrupted.
+			// m_clickCount stays 0.
+		}
+	}
```

Comment:
can probably be simplified to:
```cpp
m_clickCount = std::stoi(state.empty() ? L"0" : state.c_str());
```

Reply:
The m_clickCount is initialized next to the parameter to 0, this code is setting the State variable to be "0" if the passed state is empty (m_clickClount is already initialized), otherwise the state parameter is converted to int